File: 232/ImplementQueueUsingStacks.cs
/*
https://leetcode.com/problems/implement-queue-using-stacks/description/?envType=daily-question&envId=2024-01-29
*/

public class MyQueue {

    private Stack<int>stk1;
    private Stack<int>stk2;

    public MyQueue() {
        stk1=new Stack<int>();
        stk2=new Stack<int>();
    }
    
    public void Push(int x) {
        stk1.Push(x);
    }
    
    public int Pop() {
        while(stk2.Count!=0){
            stk2.Pop();
        }
        while(stk1.Count!=0){
            stk2.Push(stk1.Pop());
        }
        int res=stk2.Pop();
        while(stk2.Count!=0){
            stk1.Push(stk2.Pop());
        }
        return res;
    }
    
    public int Peek() {
        while(stk2.Count!=0){
            stk2.Pop();
        }
        while(stk1.Count!=0){
            stk2.Push(stk1.Pop());
        }
        int res=stk2.Peek();
        while(stk2.Count!=0){
            stk1.Push(stk2.Pop());
        }
        return res;
    }
    
    public bool Empty() {
        return stk1.Count==0;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.Push(x);
 * int param_2 = obj.Pop();
 * int param_3 = obj.Peek();
 * bool param_4 = obj.Empty();
 */

File: 2/232/ImplementQueueUsingStacks.ts
/*
https://leetcode.com/problems/implement-queue-using-stacks/description/?envType=daily-question&envId=2024-01-29
*/

class MyQueue {

    private queueList:number[];

    constructor() {
        this.queueList=[];
    }

    push(x: number): void {
        this.queueList.push(x);
    }

    pop(): number {
        let topEle:number=this.queueList[0];
        this.queueList.shift();
        return topEle;
    }

    peek(): number {
        let topEle:number=this.queueList[0];
        return topEle;
    }

    empty(): boolean {
        return this.queueList.length===0;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * var obj = new MyQueue()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.peek()
 * var param_4 = obj.empty()
 */